bot/bot.go

```diff
@@ -65,6 +67,7 @@ func (b *Bot) Run(port string) {
 	if err != nil {
 		log.Print(err)
 	}
+	b.scheduler.NewScheduler(b.server)
```

Comment:
The scheduler is just something that is part of the bot, there is no need to make a whole new class for it. Just make it a function of Bot and run it in a goroutine on start. Something like this:
```go
// scheduler: utility function to run hourly tasks
func (b *Bot) scheduler() { ... } // this will basically just be Start() but better named
```
Then in `Bot.Run()`:
```go
go b.scheduler()
```

Notice that when you were writing this you wanted access to `b.api` and `b.server`, and that `scheduler` is never needed anywhere else - doesn't it make sense to reduce its scope and just make it a function of `Bot`, since the bulk of its functionality is in the tiny 5-line `Start()`?